backend/tarsy/controllers/chat_controller.py

```
@@ -0,0 +1,486 @@
+            chat_id=chat.chat_id,
+            session_id=chat.session_id,
+            created_at_us=chat.created_at_us,
+            created_by=chat.created_by or "unknown",
+            message_count=message_count,
+        )
+
+    except ValueError as e:
+        # ChatService raises ValueError for validation failures
+        error_msg = str(e)
+        if "not found" in error_msg.lower():
+            raise HTTPException(status_code=404, detail=error_msg)
```

Ruff (0.14.3)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)